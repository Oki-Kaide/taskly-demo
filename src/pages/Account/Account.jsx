import React from 'react';
import PropTypes from 'prop-types';
import { RedButton, CompletedTask, LoggedInLayout } from '../../components';
import './Account.sass';
import '../../App.sass';

const Account = ({openConfirmModal, completedTasks, accountData, logout, error }) => {
  return (
    <LoggedInLayout mainImageUrl="./images/smiling-girl-bust.png" avatar={accountData.avatar} logout={logout}>
      <h1>
        Welcome back {accountData.name}.
      </h1>
      <div className="description">
        We are proud to say that we have completed all the tasks you requested. If you wish to continue using our service, please purchase another month.
      </div>
      { completedTasks.map((text, index) => (
        <CompletedTask text={text} key={index} />
      ))}
      <RedButton text="1 month of service" text2="5 FOOBAR" className="pay-button" handleOnClick={openConfirmModal} />
      {error ? <h4 className="error">{error}</h4> : null}
    </LoggedInLayout>
  );
}

export default Account;

Account.propTypes = {
  accountData: PropTypes.shape({}),
  logout: PropTypes.func.isRequired,
  openConfirmModal: PropTypes.func.isRequired,
  completedTasks: PropTypes.arrayOf(PropTypes.string),
}
