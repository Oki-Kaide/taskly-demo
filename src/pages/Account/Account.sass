@import '../../styles/variables.sass'

.pay-button
  margin-top: 25px
  width: 100%
  padding: 0px 20px
  display: flex
  flex-direction: row
  align-items: center
  justify-content: space-between

.account-image
  width: calc(100vw / 3) !important


