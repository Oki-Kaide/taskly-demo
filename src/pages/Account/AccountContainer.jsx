import React from 'react';
import PropTypes from 'prop-types';
import Account from './Account';
import ProtonSDK from '../../utils/proton';

class AccountContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      completedTasks: [
        'Filed your tax return',
        'Organized all your contacts',
        'Contact DMV about updating tags',
        'Sent a request to your building manager',
      ],
    };
  }

  /* istanbul ignore next */
  openConfirmModal = async () => {
    const { actor, permission, isPageHidden, setErrorState } = this.props;
    const actions = [
      {
        account: 'xtokens',
        name: 'transfer',
        authorization: [
          {
            actor: actor,
            permission: permission,
          },
        ],
        data: {
          from: actor,
          to: ProtonSDK.requestAccount,
          quantity: '5.000000 FOOBAR',
          memo: 'Taskly',
        },
      },
    ];
    const { processed, error } = await ProtonSDK.sendTransaction(actions);
    if (error) {
      setErrorState(error);
      return;
    }
    if (!processed || !processed.id) {
      return;
    }

    if (isPageHidden()) {
      window.onfocus = this.loadTasksPage;
    } else {
      this.loadTasksPage();
    }
  };

  loadTasksPage = () => {
    const { history } = this.props;
    history.push('/tasks');
    window.onfocus = null;
  };

  loadTasksPage = () => {
    const { history } = this.props;
    history.push('/tasks');
    window.onfocus = null;
  }

  render() {
    const { completedTasks } = this.state;
    const { accountData, logout, error } = this.props;

    return (
      <Account
        openConfirmModal={this.openConfirmModal}
        completedTasks={completedTasks}
        accountData={accountData}
        logout={logout}
        error={error}
      />
    );
  }
}

AccountContainer.propTypes = {
  accountData: PropTypes.shape(),
  permission: PropTypes.string,
  actor: PropTypes.string,
  logout: PropTypes.func.isRequired,
  isPageHidden: PropTypes.func.isRequired,
};

AccountContainer.defaultProps = {
  accountData: {},
  permission: '',
  actor: '',
};

export default AccountContainer;
