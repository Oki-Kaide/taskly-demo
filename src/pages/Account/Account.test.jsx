import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Account from './Account';

const createTestProps = props => ({
  openConfirmModal: jest.fn(),
  completedTasks: ['task 1', 'task 2'],
  accountData: {name: 'test', avatar:''},
  logout: jest.fn(),
  ...props,
});

describe('Account component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<Account {...props} />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<Account {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
