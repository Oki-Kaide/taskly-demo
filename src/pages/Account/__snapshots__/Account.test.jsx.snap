// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component matches the rendered snapshot 1`] = `
<LoggedInLayout
  avatar=""
  logout={[MockFunction]}
  mainImageUrl="./images/smiling-girl-bust.png"
  openConfirmModal={null}
>
  <h1>
    Welcome back 
    test
    .
  </h1>
  <div
    className="description"
  >
    We are proud to say that we have completed all the tasks you requested. If you wish to continue using our service, please purchase another month.
  </div>
  <CompletedTask
    key="0"
    text="task 1"
  />
  <CompletedTask
    key="1"
    text="task 2"
  />
  <RedButton
    className="pay-button"
    handleOnClick={[MockFunction]}
    isLoading={false}
    text="1 month of service"
    text2="5 FOOBAR"
  />
</LoggedInLayout>
`;
